cmd/dep/ensure_test.go

```diff
@@ -8,31 +8,63 @@ import (
 	"testing"
 
 	"github.com/golang/dep/internal/gps"
+	"github.com/golang/dep/internal/test"
 )
 
 func TestDeduceConstraint(t *testing.T) {
-	sv, err := gps.NewSemverConstraint("v1.2.3")
+	h := test.NewHelper(t)
+
+	cacheDir := "gps-repocache"
+	h.TempDir(cacheDir)
+	sm, err := gps.NewSourceManager(h.Path(cacheDir))
+	h.Must(err)
+
+	sv, err := gps.NewSemverConstraint("v0.8.1")
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	constraints := map[string]gps.Constraint{
-		"v1.2.3": sv,
+		"v0.8.1": sv,
+		"master": gps.NewDefaultBranch("master"),
```

Comment:
Derp! I don't know why I didn't try using `c.String() != expected.String()` to workaround `default` causing the equality check to fail. That fixes the test without having to expose NewDefaultBranch, I'll revert that change.